client/API.md

```diff
@@ -0,0 +1,536 @@
+| --------- | -------                                                                |
+| type      | The task type: `stream` or `batch`.                                    |
+| dbrps     | List of database retention policy pairs the task is allowed to access. |
+| script    | The content of the script.                                             |
+
+If any option is missing it will be left unmodified if the task already exists.
+
+#### Example
+
+```
+POST /task/TASK_NAME
+{
+    "type" : "stream",
+    "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+    "script": "stream\n    |from()\n        .measurement('cpu')\n"
+}
+```
+
+Modify only the dbrps of the task.
+
+```
+POST /task/TASK_NAME
+{
+    "dbrps": [{"db": "NEW_DATABASE_NAME", "rp" : "NEW_RP_NAME"}]
+}
+```
+
+>NOTE: Setting any DBRP will overwrite all stored DBRPs.
+
+#### Response
+
+| Code | Meaning |
+| ---- | ------- |
+| 204  | Success |
+
+
+### Get Task
+
+To get information about a task make a GET request to the `/task/TASK_NAME` endpoint.
+
+| Query Parameter | Default | Purpose                                                                                                                                                            |
+| --------------- | ------- | -------                                                                                                                                                            |
+| dot-labels      | false   | Return the DOT formatted string using only label attributes. This is useful if you intend to render a diagram from the DOT content. Otherwise it is less readable. |
+| skip-format     | false   | Do not format the TICKscript before returning.                                                                                                                     |
+
+
+#### Example
+
+Get information about a task using defaults.
+
+```
+GET /task/TASK_NAME
+```
+
+```
+{
+    "name" : "TASK_NAME",
+    "type" : "stream",
+    "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+    "script": "stream\n    |from()\n        .measurement('cpu')\n",
+    "dot" : "digraph TASK_NAME { ... }",
+    "enabled" : true,
+    "executing" : true,
+    "error" : "",
+    "stats": {}
+}
+```
+
+Get information about a task using only labels in the DOT content and skip the format step.
+
+```
+GET /task/TASK_NAME?skip-format=true&dot-labels=true
+```
+
+```
+{
+    "name" : "TASK_NAME",
+    "type" : "stream",
+    "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+    "script": "stream|from().measurement('cpu')",
+    "dot" : "digraph TASK_NAME { ... }",
+    "enabled" : true,
+    "executing" : true,
+    "error" : "",
+    "stats": {}
+}
+```
+
+#### Response
+
+| Code | Meaning             |
+| ---- | -------             |
+| 200  | Success             |
+| 404  | Task does not exist |
+
+
+### Enable Task
+
+To enable a task and start it executing make a POST request to the `/task/TASK_NAME/enable` endpoint.
```

Comment:
@goller More clarification, as per above you recommend using `PATCH` to modify an existing script instead of POST which makes sense. Here you are recommending dropping the `enable` part of the path and using `PATCH` again. 

This would mean enabling/disabling and modifying a task are all PATCH method calls to the same path. 

Does that means something like this is valid:

```
PATCH /task/TASK_NAME
{
    "script": "....",
    "enable": true
}
```

Which would modify the script and then enable the task in the same request?